src/redfishpower/redfishpower.c

```diff
@@ -868,27 +896,49 @@ static void setplugs(char **av)
 
     plugcount = hostlist_count(lplugs);
     hostindexcount = hostlist_count(hostindices);
-    if (plugcount != hostindexcount) {
-        printf("setplugs: plugs count not equal to host index count");
-        goto cleanup;
-    }
-
-    if (!(lplugsitr = hostlist_iterator_create(lplugs)))
-        err_exit(true, "hostlist_iterator_create");
-    if (!(hostindicesitr = hostlist_iterator_create(hostindices)))
-        err_exit(true, "hostlist_iterator_create");
 
     /* if user is electing to configure their own plugs, we must remove
      * all of the initial ones configured in setup_hosts()
      */
     remove_initial_plugs();
 
-    while ((plug = hostlist_next(lplugsitr))
-           && (hostindexstr = hostlist_next(hostindicesitr))) {
-        if (setup_plug(plug, hostindexstr) < 0)
+    if (plugcount != hostindexcount) {
+        /* special case, user will use plug substitution */
+        if (plugcount > 1 && hostindexcount == 1) {
+            lplugsitr = hostlist_iterator_create(lplugs);
+            if (!(hostindexstr = hostlist_nth(hostindices, 0)))
+                err_exit(false, "setplugs: hostlist_nth contains no indices");
+
+            while ((plug = hostlist_next(lplugsitr))) {
+                if (setup_plug(plug, hostindexstr) < 0) {
+                    free(plug);
+                    free(hostindexstr);
+                    goto cleanup;
+                }
+                /* multiple plugs to one host means plug substitution
+                 * required
+                 */
+                plugs_set_plug_sub_required_flag(plugs, plug, 1);
+                free(plug);
+            }
+            free(hostindexstr);
+        }
+        else {
+            printf("setplugs: plugs count not equal to host index count\n");
             goto cleanup;
-        free(plug);
-        free(hostindexstr);
+        }
+    }
+    else {
+        lplugsitr = hostlist_iterator_create(lplugs);
+        hostindicesitr = hostlist_iterator_create(hostindices);
+
+        while ((plug = hostlist_next(lplugsitr))
+               && (hostindexstr = hostlist_next(hostindicesitr))) {
+            if (setup_plug(plug, hostindexstr) < 0)
+                goto cleanup;
+            free(plug);
+            free(hostindexstr);
+        }
```

Comment:
Suggestion: use `hostlist_nth()` and a for loop since the index counts are already known.  Iterators shouldn't be needed.